### ISSUE TITLE
Bidirectional type inference for lambda produces false positive Unknown error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3000]
The following code produces a strict-mode error when it should not.

```python
# pyright: strict

from typing import TypeVar

_T = TypeVar("_T")

def func1(factory: () -> _T) -> _T:
    ...

x: set[int] = func1(lambda: set())
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.219, which I just published. It will also be in the next release of pylance.
